### ISSUE TITLE
acceptance test step can fail in CI but the pipeline passes

### ISSUE THREAD

**Issue by phil-davis** [owncloud/web#6880]
https://github.com/owncloud/web/pull/6876#issuecomment-1118222080

https://drone.owncloud.com/owncloud/web/25320/67/15

The acceptance tests step failed, but the pipeline passed. `webUIUpload/upload.feature:94` failed twice, and was reported by the drone CI steps to GitHub comment https://github.com/owncloud/web/pull/6876#issuecomment-1118188597 but the overall pipeline says it is "green" and the overall drone run passed.

Investigate what happens, particularly when `full-ci` is used and there web test fails on retry. One guess is that the code that posts the GitHub comment would have "normally" then cancelled the whole drone job (which would make it fail). But because full-ci is used, the drone job cancel gets skipped. The step that posts the GitHub comment probably then exits with "normal" status, and so drone sees the "normal/success" status of the step and thinks that the pipeline is OK.

Needs to be sorted out, otherwise we might be merging `full-ci` runs that actually have some test fails.

**Comment by phil-davis** [comment]
If/when a drone starlark code change is made, also check other repos, and if they have a similar problem then raise a general issue in `owncloud/qa` to get the fix applied "everywhere".

**Comment by saw-jan** [comment]
@phil-davis I am confused. You mentioned that the test failed but the pipeline passed in the build https://drone.owncloud.com/owncloud/web/25320/67/15 but the pipeline has actually failed.

**Comment by phil-davis** [comment]
It looks like I was confused! The drone CI has run 4 times:
https://drone.owncloud.com/owncloud/web/25319 - that was the first run that had the unexpected passes:
```
runsh: Total unexpected passed scenarios throughout the test run:
webUIFilesSearch/search.feature:50
webUIFilesSearch/search.feature:129
runsh: Exit code: 1
```
Then https://drone.owncloud.com/owncloud/web/25320
That had the fail of `webUIUpload/upload.feature:94` which posted the GitHub comment. There is no longer any link to that from the commit list on the PR, because drone has been restarted on the same commit. The whole drone job did fail - correct.

Next https://drone.owncloud.com/owncloud/web/25321
That was a rerun of 25320 and passed, which made the GitHub PR show green. I think that I got confused at that point, I saw the comment pointing to the failed pipeline, but also saw that the CI was green.

I restarted CI again (thinking that there was this problem) https://drone.owncloud.com/owncloud/web/25323
That also passed.

So it was my confusion! Closing this issue.
